### PR TITLE
Removed unneccessary require commands from provider_arm.

### PR DESCRIPTION
**Pull request by chrismatteson**
Added classes and templates to install Azure dependancies.  Additionally updated metadata.json for dependancies on stdlib

### CODE REVIEW

File: spec/acceptance/azure_setup_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'spec_helper_acceptance'
+
+
+describe 'azure class' do
+  azurepackages = ['retries', 'azure', 'azure_mgmt_compute', 'azure_mgmt_storage','azure_mgmt_resources', 'azure_mgmt_network', 'hocon']
+
+#  case fact('osfamily')
```
Comment (gregohardy) [inline spec/acceptance/azure_setup_spec.rb]: Cool. You pass in the packages based on osfamily in the pararms. This isn't needed is it?

File: spec/classes/azure_spec.rb [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      let(:params) do
+        {
+          :subscription_id => 'fake',
+          :tenant_id => 'fake',
+          :client_id => 'fake',
+          :client_secret => 'fake'
+        }
+      end
+# Commented out due to error with path validation
+#      it { is_expected.to compile.with_all_deps }
```
Comment (gregohardy) [inline spec/classes/azure_spec.rb]: What path is not validating? Anything we can help with?

File: spec/classes/azure_spec.rb [hunk 3]
```diff
@@ -0,0 +1,206 @@
+require 'spec_helper'
+
+describe 'azure' do
```
Comment (gregohardy) [inline spec/classes/azure_spec.rb]: Naming. :)

File: spec/classes/azure_spec.rb [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+          :client_id => 'fake',
+          :client_secret => 'fake'
+        }
+      end
+      it { is_expected.to compile.with_all_deps }
+      it { should contain_file('/etc/puppetlabs/puppet/azure.conf') }
+    end
+  end
+
+  context 'Debian' do
```
Comment (gregohardy) [inline spec/classes/azure_spec.rb]: Again. Naming.

**Comment by gregohardy** [comment]
The unit tests are great. I just ran them here. Good job.

File: spec/acceptance/azure_setup_spec.rb (hunk 1 above)
Comment (chrismatteson) [inline spec/acceptance/azure_setup_spec.rb]: There are two sets of packages, the gem packages that every OS needs, and the dependencies to compile nokogiri, which vary based on OS.  I never got the acceptance tests running correctly, hence commenting out the case statement here.  Need to tweak it more but the hardcoded 2015.2 version seems like the nodesets are busted to begin with.

File: spec/classes/azure_spec.rb (hunk 2 above)
Comment (chrismatteson) [inline spec/classes/azure_spec.rb]: Apparently if you run spec tests on linux for windows it'll complain that the path is invalid.  There is a pull request in process to fix this: https://github.com/rodjek/rspec-puppet/pull/481

File: spec/classes/azure_spec.rb (hunk 3 above)
Comment (chrismatteson) [inline spec/classes/azure_spec.rb]: So green behind the ears here that I have no idea what I named wrong :D

Comment (chrismatteson) [inline spec/classes/azure_spec.rb]: Mixed up a couple different sayings here but you get the point...  Apparently I'm not the first to make that mistake though: http://languagelog.ldc.upenn.edu/nll/?p=727

File: spec/classes/azure_spec.rb (hunk 2 above)
Comment (chrismatteson) [inline spec/classes/azure_spec.rb]: Specifically the c:\ProgramData... etc path in the File resource for the azure.conf file.

File: spec/classes/azure_spec.rb (hunk 3 above)
Comment (gregohardy) [inline spec/classes/azure_spec.rb]: No, you are grand! Just the : describe 'Need a descriptive tag here about the test' :)

File: spec/classes/azure_spec.rb (hunk 2 above)
Comment (gregohardy) [inline spec/classes/azure_spec.rb]: That makes sense. Okay. Im running the acceptance tests now on centos7 to see what needs to be done.

**Comment by gregohardy** [comment]
Hi Chris. I tested the classes on centos7 today and they worked great. Can we squash this and remove any commented code etc?

**Comment by gregohardy** [comment]
Unfortunately Chris has left Puppet. Closing this.

**Pull request closed without merging**